front/src/applications/operationalStudies/hooks.ts

```diff
@@ -0,0 +1,133 @@
+  const trainScheduleIDsToModify = useSelector(getTrainScheduleIDsToModify);
+  const usingElectricalProfiles = useSelector(getUsingElectricalProfiles);
+  const dispatch = useAppDispatch();
+  const osrdActions = useOsrdConfActions();
+
+  const [getTrainScheduleById] = osrdEditoastApi.endpoints.getV2TrainScheduleById.useLazyQuery({});
+  const [getRollingStockByName] =
+    osrdEditoastApi.endpoints.getRollingStockNameByRollingStockName.useLazyQuery();
+  const [postPathfindingBlocks] =
+    osrdEditoastApi.endpoints.postV2InfraByInfraIdPathfindingBlocks.useMutation();
+  const [postPathProperties] =
+    enhancedEditoastApi.endpoints.postV2InfraByInfraIdPathProperties.useMutation();
+
+  useEffect(() => {
+    const setupItineraryForTrainUpdate = async () => {
+      const trainSchedule = await getTrainScheduleById({
+        id: trainScheduleIDsToModify[0],
+      }).unwrap();
+
+      if (trainSchedule.rolling_stock_name && infraId) {
+        const rollingStock = await getRollingStockByName({
+          rollingStockName: trainSchedule.rolling_stock_name,
+        }).unwrap();
+
+        // TODO TS2 : Next part might not be needed (except to updePathSteps), we need inly trainSchedulePath and
+        // rolling stock infos to relaunch the pathfinding. Check for that in simulation results issue
+
+        const params: PostV2InfraByInfraIdPathfindingBlocksApiArg = {
+          infraId,
+          pathfindingInputV2: {
+            path_items: trainSchedule.path,
+            rolling_stock_is_thermal: isThermal(rollingStock.effort_curves.modes),
+            rolling_stock_loading_gauge: rollingStock.loading_gauge,
+            rolling_stock_supported_electrifications: getSupportedElectrification(
+              rollingStock.effort_curves.modes
+            ),
+            rolling_stock_supported_signaling_systems: rollingStock.supported_signaling_systems,
+          },
+        };
+
+        try {
+          const pathfindingResult = await postPathfindingBlocks(params).unwrap();
+          if (pathfindingResult.status === 'success') {
+            const pathPropertiesParams: PostV2InfraByInfraIdPathPropertiesApiArg = {
+              infraId,
+              props: ['electrifications', 'geometry', 'operational_points'],
+              pathPropertiesInput: {
+                track_section_ranges: pathfindingResult.track_section_ranges,
+              },
+            };
+            const { electrifications, geometry, operational_points } =
+              await postPathProperties(pathPropertiesParams).unwrap();
+
+            if (electrifications && geometry && operational_points) {
+              const suggestedOperationalPoints: SuggestedOP[] = formatSuggestedOperationalPoints(
+                operational_points,
+                geometry,
+                pathfindingResult.length
+              );
+              // TODO TS2 : update operational_points property with vias added on map included in trainSchedule.path
```

Comment:
Directly insert the vias (like in PathfindingV2)
```suggestion
              cons suggestedOPsWithVias = insertViasInOPs(
                suggestedOperationalPoints,
                pathSteps
              );
```

Reply:
Done